File: Codigo/Ejercicio 3/Try.cpp
#include <iostream>
#include <stdexcept> // Necesario para std::invalid_argument

int main() {
    try {
        std::string cadena;
        std::cin >> cadena;
        int numero = std::stoi(cadena); // Intenta convertir la cadena en un entero

        // Si la conversión tiene éxito, imprime el número
        std::cout << "Número convertido: " << numero << std::endl;
    } catch (const std::invalid_argument& e) {
        // Captura una excepción si la conversión falla debido a un argumento no válido
        std::cerr << "Error de conversión: " << e.what() << std::endl;
    } catch (const std::out_of_range& e) {
        // Captura una excepción si la conversión falla debido a un desbordamiento de rango
        std::cerr << "Error de rango: " << e.what() << std::endl;
    }

    return 0;
}

File: Codigo/Ejercicio 3/Try.java
import java.util.Scanner;

public class EjemploTryCatch2 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        try {
            System.out.print("Ingrese un número: ");
            int numero = sc.nextInt(); // Intenta leer un entero del usuario

            System.out.println("Número ingresado: " + numero);
        } catch (java.util.InputMismatchException e) {
            System.err.println("Error: Ingrese un número válido.");
        }

        sc.close();
    }
}
